### PR TITLE
Add hunspell/1.7.0

### PR DESCRIPTION
**Pull request by friendlyanon**
Specify library name and version:  **hunspell/1.7.0**

Issue: #8618

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/hunspell/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
+        # NOTE: The source contains a pre-configured hunvisapi.h and it would
+        #       prevent no_copy_source and building without patches.
+        h = os.path.join(self.source_folder, "src", "hunspell", "hunvisapi.h")
+        os.remove(h)
```
Comment (intelligide) [inline recipes/hunspell/all/conanfile.py:46]:
```suggestion
      os.remove(os.path.join(self.source_folder, "src", "hunspell", "hunvisapi.h"))
```

**Review by intelligide — CHANGES_REQUESTED** [review]

File: recipes/hunspell/all/conanfile.py (hunk 1 above)
Comment (friendlyanon) [reply recipes/hunspell/all/conanfile.py:46]: PEP 8 requires 79 long lines.

File: recipes/hunspell/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+class HunspellConan(ConanFile):
+    name = "hunspell"
+    description = (
+        "Hunspell is a free spell checker and morphological analyzer library"
+    )
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://hunspell.github.io/"
+    topics = ("spell", "spell-check")
+    license = "???"
```
[the remark this answers is not in the document]
Comment (friendlyanon) [reply recipes/hunspell/all/conanfile.py:18]: I was going to start a discussion on the license. The project has 3 licenses. What's the most appropriate to select in these cases? What's the rationale that you used to select LGPL?

File: recipes/hunspell/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    settings = ("os", "arch", "compiler", "build_type")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    generators = ("cmake",)
```
[the remark this answers is not in the document]
Comment (friendlyanon) [reply recipes/hunspell/all/conanfile.py:28]: What's the significance of this?

File: recipes/hunspell/all/conanfile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (intelligide) [reply recipes/hunspell/all/conanfile.py:18]: `license` can be a tuple. :wink:

File: recipes/hunspell/all/conanfile.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (intelligide) [reply recipes/hunspell/all/conanfile.py:28]: `generators` can be a tuple or a simple string. It is for more simplicity and readability if there is only one generator.

File: recipes/hunspell/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+
+class TestPackageConan(ConanFile):
+    settings = ("os", "compiler", "build_type", "arch")
+    generators = ("cmake", "cmake_find_package_multi")
```
[the remark this answers is not in the document]
Comment (intelligide) [reply recipes/hunspell/all/test_package/conanfile.py:8]: conanfile attributes that are tuples also support simple elements without having to be in a tuple. So, `generators = ("cmake",)` and `generators = "cmake"` produce the same result. We use as few parentheses as possible to keep the recipes as simple and readable as possible.

**Review by toge — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/hunspell/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
+import functools
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
```
Comment (uilianries) [inline recipes/hunspell/all/conanfile.py:5]:
```suggestion
```

Ununsed

**Review by uilianries — APPROVED** [review]

**Pull request merged**